src/swap/utils/processSwapTransaction.ts

```diff
@@ -105,32 +84,30 @@ export async function processSwapTransaction({
   }
 
   // make the swap
-  setLifeCycleStatus({
+  updateLifeCycleStatus({
     statusName: 'transactionPending',
-    statusData: {
-      // LifecycleStatus shared data
-      isMissingRequiredField: lifeCycleStatus.statusData.isMissingRequiredField,
-      maxSlippage: lifeCycleStatus.statusData.maxSlippage,
-    },
   });
   const txHash = await sendTransactionAsync({
     to: transaction.to,
     value: transaction.value,
     data: transaction.data,
   });
-
+  updateLifeCycleStatus({
+    statusName: 'transactionApproved',
+    statusData: {
+      transactionHash: txHash,
+      transactionType: useAggregator ? 'ERC20' : 'Permit2',
+    },
+  });
```

Comment:
Currently we are showing confirm in wallet until the receipt, seemed like we should emit the transaction status and then you'll get success after receipt.

Reply:
I lean towards your point, but can we ask also @0xAlec @cpcramer @mykcryptodev @abcrane123 what they think about this.

Reply:
+1